### PR TITLE
Introduce new losses (part 1)

### PR DESCRIPTION
**Pull request by cofri**
This PR introduces new losses and new callbacks:

- two new losses `TauCategoricalCrossentropy` and `CategoricalHinge`
- callbacks `LossParamScheduler` to modify loss hyper-parameters during training and `LossParamLog` to log the value of parameters

Some changes have also been brought:
- the centered hinge losses (`HingeMargin` and `MulticlassHinge`) have now a `min_margin/2` on both sides
- the loss hyper-parameters `min_margin`, `alpha` and `tau` are now `tf.Variable` in order to modify them during training

### CODE REVIEW

**Review by franckma31 — CHANGES_REQUESTED** [review]
Need to verify and correct class CategoricalHinge(Loss):
neg = tf.reduce_max((1.0 - y_true) * y_pred, axis=-1)
if y_pred <0 for non target output neg will be set to 0
May be also add a unitary test to check this property

**Review by franckma31 — APPROVED** [review]
All fine!
